File: src/main/kotlin/wowsft/model/gameparams/ship/component/engine/Engine.java
package wowsft.model.gameparams.ship.component.engine;

import wowsft.config.WoWSFT;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
public class Engine
{
    private float backwardEngineForsag;
    private float backwardEngineForsagMaxSpeed;
    private float backwardEngineUpTime;
    private float backwardSpeedOnFlood;
    private float damagedEnginePowerMultiplier;
    private float damagedEnginePowerTimeMultiplier;
    private float forwardEngineForsag;
    private float forwardEngineForsagMaxSpeed;
    private float forwardEngineUpTime;
    private float forwardSpeedOnFlood;
    private float histEnginePower;
    private float speedCoef;
}

File: src/main/kotlin/wowsft/model/gameparams/ship/component/engine/Engine.kt
package wowsft.model.gameparams.ship.component.engine

import wowsft.config.WoWSFT
import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class Engine {
    var backwardEngineForsag = 0.toDouble()
    var backwardEngineForsagMaxSpeed = 0.toDouble()
    var backwardEngineUpTime = 0.toDouble()
    var backwardSpeedOnFlood = 0.toDouble()
    var damagedEnginePowerMultiplier = 0.toDouble()
    var damagedEnginePowerTimeMultiplier = 0.toDouble()
    var forwardEngineForsag = 0.toDouble()
    var forwardEngineForsagMaxSpeed = 0.toDouble()
    var forwardEngineUpTime = 0.toDouble()
    var forwardSpeedOnFlood = 0.toDouble()
    var histEnginePower = 0.toDouble()
    var speedCoef = 0.toDouble()
}
